src/storage/mvcc/txn.rs

```diff
@@ -594,6 +600,73 @@ impl<S: Snapshot> MvccTxn<S> {
         })
     }
 
+    /// Check the status of a transaction.
+    ///
+    /// This operation checks whether a transaction has expired it's Lock's TTL, rollback the
+    /// transaction if expired, and update the transaction's min_commit_ts according to the metadata
+    /// in the primary lock.
+    ///
+    /// When transaction T1 meets T2's lock, it may invoke this on T2's primary key. In this
+    /// situation, `self.start_ts` is T2's `start_ts`, `caller_start_ts` is T1's `start_ts`, and
+    /// the `current_ts` is literally the timestamp when this function is invoked. It may not be
+    /// accurate.
+    ///
+    /// Returns (`lock_ttl`, `commit_ts`, `is_pessimistic_txn`).
+    /// After checking, if the lock is still alive, it retrieves the Lock's TTL; if the transaction
+    /// is committed, get the commit_ts; otherwise, if the transaction is rolled back or there's
+    /// no information about the transaction, results will be both 0.
+    pub fn check_txn_status(
+        &mut self,
+        primary_key: Key,
+        caller_start_ts: u64,
+        current_ts: u64,
+    ) -> Result<(u64, u64, bool)> {
```

Comment:
How about returning `Result<Option<TXNStatus>>`?

Reply:
I'm going to add a struct TxnStatus which contains `lock_ttl` and `commit_ts` in a further PR. But this function may returns `Result<(TxnStatus, bool)>` then.